### PR TITLE
Data science

### PR DESCRIPTION
**Pull request by elialbert**
all changes made, chi-squared values verified externally.

### CODE REVIEW

**Comment by elialbert** [comment]
woops lemme fix hound

**Comment by elialbert** [comment]
ok fixed

File: data_science/optimizer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  describe "calculations" do
+    it "should calculate conversion rates" do
+      result = optimizer.conversion_rate_for(:A)
+      expect(result[0]).to be_within(0.1).of(0.24527)
+      expect(result[0]).to be < result[1]
+    end
+
+    it "should do chisquared calculations" do
+      puts optimizer.result_confidence
```
Comment (zsyed1) [inline data_science/optimizer_spec.rb]: Remove `puts` :smile:

File: data_science/optimizer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    end
+
+    it "counts number of conversions per cohort" do
+      expect(optimizer.simple_counts[:A][:conversions]).to eq(12)
+    end
+  end
+
+  describe "calculations" do
+    it "should calculate conversion rates" do
+      result = optimizer.conversion_rate_for(:A)
```
Comment (zsyed1) [inline data_science/optimizer_spec.rb]: Should be in a `let`

File: data_science/test_data.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  {"cohort": "B", "result": 0},
+  {"cohort": "A", "result": 1}, 
+  {"cohort": "A", "result": 0}, 
+  {"cohort": "B", "result": 0},
+  {"cohort": "A", "result": 1}, 
+  {"cohort": "A", "result": 0}, 
+  {"cohort": "B", "result": 1},
+  {"cohort": "A", "result": 0}, 
+  {"cohort": "A", "result": 0}, 
+  {"cohort": "B", "result": 0}]
```
Comment (zsyed1) [inline data_science/test_data.json]: Very nice - this is closed!

**Pull request closed without merging**